pkgs/development/guile-modules/guile-sdl2/default.nix

```diff
@@ -1,46 +1,38 @@
-{ stdenv,
-  SDL2,
-  SDL2_image,
-  SDL2_ttf,
-  SDL2_mixer,
-  autoconf,
-  automake,
-  fetchgit,
-  guile,
-  libtool,
-  pkgconfig
-  }:
-stdenv.mkDerivation rec {
-  name = "guile-sdl2-${version}";
-  version = "0.1.0";
-  buildInputs = [ autoconf
-                  automake
-                  SDL2
-                  SDL2_image
-                  SDL2_ttf
-                  SDL2_mixer
-                  libtool
-                  guile
-                  pkgconfig ];
-  src = fetchgit {
-    url = "git://dthompson.us/guile-sdl2.git";
-    rev = "048f80ddb5c6b03b87bba199a99a6f22d911bfff";
-    sha256 = "1v7bc2bsddb46qdzq7cyzlw5i2y175kh66mbzbjky85sjfypb084";
+{ stdenv, fetchurl, guile, libtool, pkgconfig
+, SDL2, SDL2_image, SDL2_ttf, SDL2_mixer
+}:
+
+let
+  name = "${pname}-${version}";
+  pname = "guile-sdl2";
+  version = "0.2.0";
```

Comment:
Could you bring these into `mkDerivation` for consistency with other derivations?

Reply:
The reason I put it outside so that `mkDerivation` doesn't need `rec`. Which other derivations are you talking about?

Reply:
According to this approximate count, most of them:
```
$ ag -G '\.nix$' -l '(?s)version =.*mkDerivation' | wc -l
678
$ ag -G '\.nix$' -l '(?s)mkDerivation rec .*version =' | wc -l
3494
```
and, in particular, the previous PR #29886. There is no need to remove rec from mkDerivation.

Reply:
But wouldn't you say it's better without rec? I'm bit hesitant because this would mean it's inconsistent with most other guile-modules that do not need to use rec: https://github.com/NixOS/nixpkgs/commit/3579d7e2b2fb7befcbebc473115c230d3c0781c0

Reply:
I guess my point is why use rec when it's not needed?

Reply:
#29886 doesn't remove rec, in fact it needs it.

Reply:
I think that in general these attrs are put into mkDerivation because `rec` version is shorter, and they can be used from other derivations — for example, grep `.version}` in the repository.

Reply:
Oh that seems really useful, I didn't think about that.

So I adopted this style after seeing it somewhere, mainly just because of `version` and how putting it in `let` can remove the need for `rec` in a lot of cases. And otherwise if this was not done, `rec` seemed to nearly always be needed, which didn't seem right.

Anyway in that case I'll switch to setting `version` too in `mkDerivation`, thanks for mentioning this. :)

Reply:
I do not have strong feelings about the `rec` but just so you know:  there is a function called `builtins.parseDrvName` to get version info from derivation name.  Which makes putting version into the derivation a second time not necessary.  Because `parseDrvName` should work with far more derivations than asking for a `version` attribute using the builtin function would be more desirable anyway, at least in my opinion.

Reply:
@seppeljordan Oh right that's even better, thanks for letting me know. :)